### PR TITLE
feat: Update "Add Members" to Workspace experience

### PR DESCRIPTION
**Pull request by gt2345**
## Description

Update "Add Members" to Workspace experience according to WEB-1740
Also migrate Select at WorkspaceMemberAddModalComponent to use the Select component in UI kit.
Also changed the `assignRolesToGroup` function to align with `assignRolesToUser` to accept list of payload.

## Test Plan

With RBAC enabled, navigate to workspace details page, then members tab.
Click the button "Add Members", all labels should be as described in the ticket.
Button "Add to workspace" should be disabled.
At `Find and Select Users or Groups` select, you should not see any option before typing anything.
After typing, you should see options of groups and users with name containing the search phrase, case insensitive. 
Select multiple users and/or groups.
Button "Add to workspace" should be enabled.
Click on the button, selected user and/or groups should be added to the workspace. 

Screenshot:
Screenshot 2023-10-18 at 4 40 35 PM

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by keita-determined** [comment]
@gt2345 could you resolve conflicts to run it locally

File: webui/react/src/components/WorkspaceMemberAddModal.tsx [hunk 1]
```diff
@@ -118,55 +104,69 @@ const WorkspaceMemberAddModalComponent: React.FC<Props> = ({
...
+            mode="multiple"
+            optionLabelProp="label"
+            placeholder="Add Users or Groups"
+            ref={ref}
+            searchable={true}
+            onBlur={() => setFilteredOption([])}
+            onSearch={handleSearch}>
+            {_.map(filteredOption, (option) => (
+              <Option
+                key={(isUser(option) ? 'u_' : 'g_') + getIdFromUserOrGroup(option)}
```
Comment (keita-determined) [inline webui/react/src/components/WorkspaceMemberAddModal.tsx:143]: we use `u_` and `g_` in many places. these can be constant values

File: webui/react/src/components/WorkspaceMemberAddModal.tsx [hunk 2]
```diff
@@ -19,84 +21,68 @@ interface Props {
...
-              scopeWorkspaceId: workspaceId,
-            });
+      if (values) {
+        const userOrGroup = _.groupBy(values.userOrGroupId, (o) => o.substring(0, 2));
+        const groupPayload = _.map(userOrGroup['g_'], (o) => ({
+          groupId: Number(o.substring(2)),
+          roleIds: [values.roleId],
+          scopeWorkspaceId: workspace.id,
+        }));
+        groupPayload.length > 0 && (await assignRolesToGroup(groupPayload));
```
Comment (keita-determined) [inline webui/react/src/components/WorkspaceMemberAddModal.tsx:72]: nit: it's probably personal preference but i think if statement is more readable than `&&`, but up to you

File: webui/react/src/components/WorkspaceMemberAddModal.tsx [hunk 3]
```diff
@@ -19,84 +21,68 @@ interface Props {
...
   rolesAssignableToScope,
   onClose,
-  workspaceId,
+  workspace,
 }: Props) => {
   const idPrefix = useId();
-  const [selectedOption, setSelectedOption] = useState<UserOrGroup>();
+  const [filteredOption, setFilteredOption] = useState<UserOrGroup[]>([]);
   const [form] = Form.useForm<FormInputs>();
+  const ref = useRef<RefSelectProps>(null);
```
Comment (keita-determined) [inline webui/react/src/components/WorkspaceMemberAddModal.tsx:40]: is it `RefSelectProps | null`?

File: webui/react/src/components/WorkspaceMemberAddModal.tsx [hunk 4]
```diff
@@ -118,55 +104,69 @@ const WorkspaceMemberAddModalComponent: React.FC<Props> = ({
...
-          />
+            filterOption={false}
+            mode="multiple"
+            optionLabelProp="label"
+            placeholder="Add Users or Groups"
+            ref={ref}
+            searchable={true}
+            onBlur={() => setFilteredOption([])}
+            onSearch={handleSearch}>
+            {_.map(filteredOption, (option) => (
```
[the remark this answers is not in the document]
Comment (gt2345) [reply webui/react/src/components/WorkspaceMemberAddModal.tsx:141]: I don't think js `map` can take objects that are not array?

**Review by keita-determined — COMMENTED** [review]
ran on local, and got some findings

File: webui/react/src/components/WorkspaceMemberAddModal.tsx [hunk 5]
```diff
@@ -19,84 +19,68 @@ interface Props {
...
 const WorkspaceMemberAddModalComponent: React.FC<Props> = ({
   addableUsersAndGroups,
   rolesAssignableToScope,
   onClose,
-  workspaceId,
+  workspace,
 }: Props) => {
   const idPrefix = useId();
-  const [selectedOption, setSelectedOption] = useState<UserOrGroup>();
+  const [filteredOption, setFilteredOption] = useState<UserOrGroup[]>([]);
```
[the remark this answers is not in the document]
Comment (gt2345) [reply webui/react/src/components/WorkspaceMemberAddModal.tsx:40]: I believe this is one of the asks from the ticket

> Update logic to only show list when there is user input for a new selection (aka users never see the full list, only see the filtered list after typing a character)

[the remark this answers is not in the document]
Comment (keita-determined) [reply webui/react/src/components/WorkspaceMemberAddModal.tsx:40]: i see. I missed that part. thats interesting behavior to me, but ok

**Review by keita-determined — APPROVED** [review]
LGTM

**Pull request merged**